### ISSUE TITLE
Sending SMS from Hubspot Workflows (Automation)

### ISSUE THREAD

**Issue by dochollywood8** [ringcentral/hubspot-embeddable-ringcentral-phone#162]
Feature request -
Is it possible to send out SMS messages through Hubspot workflows (automation)?

**Comment by zxdong262** [comment]
@dochollywood8 Hi, can you provide more detail, I am not familiar with Hubspot workflow.

**Comment by dochollywood8** [comment]
Hubspot workflows are the part if the Hubspot platform that enables
marketing and sales automation.

For example, workflows allow a business to enroll all contacts who meet
specific criteria (for example, all contacts who are new signups to GitHub
if Github were the business) into an automated workflow, such as sending
out specific emails or text messages at specific times.

One of Hubspots missing features is that it does not have SMS messaging
built in to the platform, so using a third party platform that interested
with Hubspot (such as SalesMsg) is needed to send out SMS whether automated
through workflows or manually.

It would be very helpful for the RC Hubspot integration to allow this.



Larry Fan, MD

**Comment by zxdong262** [comment]
@dochollywood8 Thank you for the feedback, I will look into this, seems interesting!

**Comment by dochollywood8** [comment]
Yes, you can see there platforms Just Call, Kixie, and Simplii/Fusion with
more robust Hubspot integrations including workflows, call dispositioning,
and voicemail drops.


Larry Fan, MD

**Comment by codefix25** [comment]
yes I am also requesting such functionality for SMS integration in Workflows, as well as voicemail drops.
In addition the functionality requirements should be as follows:
1. Ability to set times/days when SMS go out - in other words they should be held until for example work hours 9 am - 5 pm. User defined preference.
2. Unenroll feature support (opt-out/resubscribe). If the user responds with "stop" or "unenroll", SMS should be stopped. 
3. In general should replicate / support features similar to Sakari SMS,  SMSZap, SalesMsg apps.  https://www.smszap.com/features
- ability to use Hubspot custom fields "firstname", lastname, etc within SMS 
- SMS templates
- ability to create new contacts in Hubspot from incoming SMS if they are not in the database based on the phone number.

https://legacydocs.hubspot.com/docs/methods/workflows/workflows_overview
https://legacydocs.hubspot.com/docs/methods/workflow-extensions/workflow-extensions-overview

**Comment by zxdong262** [comment]
@codefix25 @dochollywood8 Now we have a RingCentral integration in HubSpot app market you can use, it support SMS in workflow.
